core/corehttp/gateway_handler.go

```diff
@@ -271,120 +270,61 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	logger := log.With("from", r.RequestURI)
 	logger.Debug("http request received")
 
-	// X-Ipfs-Gateway-Prefix was removed (https://github.com/ipfs/go-ipfs/issues/7702)
-	// TODO: remove this after  go-ipfs 0.13 ships
-	if prfx := r.Header.Get("X-Ipfs-Gateway-Prefix"); prfx != "" {
-		err := fmt.Errorf("X-Ipfs-Gateway-Prefix support was removed: https://github.com/ipfs/go-ipfs/issues/7702")
-		webError(w, "unsupported HTTP header", err, http.StatusBadRequest)
+	if !validateSupportedHeaders(w, r) {
```

Comment:
I extracted some smaller functions from the code, partly to make it easier to reason about and partly because I had to extract `getOrHeadHandlerUnixfs` out of `getOrHeadHandler`, which needed to reuse some of this logic.

That said, I'm new to Go and coming most recently from an Elixir background, so I'm guessing I'm not writing idiomatic Go.  Please don't hesitate to suggest changes, or function renames or whatever.

Reply:
Excellent, that does sound nicer and more consistent.  I'll update all of these to simpler names with no negation.